From cbe50e0e86c7e44f4106ab00ead0a06fdba6e02a Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Mon, 8 Feb 2021 02:43:16 -0800
Subject: [PATCH 1252/1921] arch/arm64: dts: cn9130-crb fix the SD GPIOs

The SD GPIOs on CN9130-CRB board are connected to CP GPIO pins
and not to i2c expander.

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I688bc6a3eb394f7c30640769c48563c7b4c0081d
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/45530
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-crb.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-crb.dts b/arch/arm64/boot/dts/marvell/cn9130-crb.dts
index 542570b51309..b5d3e4f98d6c 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-crb.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-crb.dts
@@ -58,7 +58,7 @@
 		regulator-name = "cp0_sd_vccq";
 		regulator-min-microvolt = <1800000>;
 		regulator-max-microvolt = <3300000>;
-		gpios = <&expander0 15 GPIO_ACTIVE_HIGH>;
+		gpios = <&cp0_gpio2 18 GPIO_ACTIVE_HIGH>;
 		states = <1800000 0x1
 			  3300000 0x0>;
 	};
@@ -68,7 +68,7 @@
 		regulator-name = "cp0_sd_vcc";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
-		gpio = <&expander0 14 GPIO_ACTIVE_HIGH>;
+		gpio = <&cp0_gpio2 19 GPIO_ACTIVE_HIGH>;
 		enable-active-high;
 		regulator-always-on;
 	};
-- 
2.31.1

